sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/ShuffledBatchRDD.scala

```diff
@@ -155,14 +155,17 @@ class ShuffledBatchRDD(
   }
 
   override def compute(split: Partition, context: TaskContext): Iterator[ColumnarBatch] = {
+    metrics(GpuMetricNames.NUM_OUTPUT_PARTITIONS).add(1)
+
     val shuffledRowPartition = split.asInstanceOf[ShuffledBatchRDDPartition]
     val tempMetrics = context.taskMetrics().createTempShuffleReadMetrics()
     // `SQLShuffleReadMetricsReporter` will update its own metrics for SQL exchange operator,
     // as well as the `tempMetrics` for basic shuffle metrics.
     val sqlMetricsReporter = new SQLShuffleReadMetricsReporter(tempMetrics, metrics)
     val shuffleManagerShims = ShimLoader.getSparkShims.getShuffleManagerShims()
-    val reader = split.asInstanceOf[ShuffledBatchRDDPartition].spec match {
+    val reader = shuffledRowPartition.spec match {
```

Comment:
This change was not strictly necessary but I thought it made sense to use the variable that was previously unused.